### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by garrensmith**
Moves the errors into their own file. To reduce the file size a bit. I've added a test that fails on the original mobc but passes with our fixes. Also removed the `Strategy`. We don't use that anymore

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+impl<E> fmt::Display for Error<E>
+where
+    E: fmt::Display + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Inner(ref err) => write!(f, "{}", err),
+            Error::Timeout => write!(f, "Timed out in mobc"),
```
Comment (pimeys) [inline src/error.rs:28]: Maybe `Timed out in the connection pool`. Our users don't need to know the name of the library.

File: src/error.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            Error::BadConn => write!(f, "Bad connection in mobc"),
+            Error::PoolClosed => write!(f, "The pool is closed"),
+        }
+    }
+}
+
+impl<E> fmt::Debug for Error<E>
```
Comment (pimeys) [inline src/error.rs:35]: Do we really need this?

File: src/lib.rs [hunk 3]
```diff
@@ -307,12 +254,6 @@ impl fmt::Debug for State {
     }
 }
 
-#[derive(PartialEq)]
```
Comment (pimeys) [inline src/lib.rs:310]: 💯

File: tests/mobc.rs [hunk 4]
```diff
@@ -1134,3 +1135,54 @@ fn test_requests_skip_canceled() {
     })
     .unwrap();
 }
+
+#[cfg(all(
```
Comment (pimeys) [inline tests/mobc.rs:1139]: 💯

**Review by pimeys — APPROVED** [review]

File: src/error.rs (hunk 2 above)
Comment (garrensmith) [reply src/error.rs:35]: Probably not. But I'm going to leave it there for now.

Comment (garrensmith) [reply src/error.rs:35]: Looking at it, you are correct. I'm going to delete.

Comment (garrensmith) [reply src/error.rs:35]: Ah, its needed for the `impl<E> error::Error for Error<E>` below.

File: src/error.rs [hunk 5]
```diff
@@ -0,0 +1,61 @@
+use std::error;
+use std::fmt;
+/// The error type returned by methods in this crate.
+pub enum Error<E> {
```
Comment (pimeys) [inline src/error.rs:4]: Maybe this could be easier to do just by using `thiserror` crate?

Comment (garrensmith) [reply src/error.rs:4]: Nice that worked great

**Pull request merged**